### PR TITLE
TWI-11 Get current profile

### PR DESCRIPTION
**Pull request by pm888**

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -18,7 +18,7 @@ func Server() {
 	r.HandleFunc("/v1/users/create", Serviceuser.CreateUser).Methods(http.MethodPost)
 	r.HandleFunc("/v1/users/login", Serviceuser.LoginUsers).Methods(http.MethodPost)
 	http.Handle("/v1/users/logout", Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.LogoutUser)))
-	http.Handle("/v1/users/{id}", Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.GetCurrentProfile)))
+	r.HandleFunc("/v1/users/{id}", Serviceuser.GetCurrentProfile).Methods(http.MethodGet)
```
Comment (bunin) [inline internal/server/server.go]: why removing Auth middleware, if it's really "current profile" it has to have the authorized user

**Review by bunin — CHANGES_REQUESTED** [review]

File: openapi.yaml [hunk 2]
```diff
@@ -94,12 +94,12 @@ paths:
         - Users
       summary: Get Current User Profile
       parameters:
-        - name: id
-          in: path
-          description: User ID
+        - name: X-API-KEY
```
Comment (bunin) [inline openapi.yaml]: this should be in the auth/security section

**Review by bunin — CHANGES_REQUESTED** [review]

File: internal/server/server.go [hunk 3]
```diff
@@ -17,10 +17,10 @@ func Server() {
 	r.Use(CorsMiddleware)
 	r.HandleFunc("/v1/users/create", Serviceuser.CreateUser).Methods(http.MethodPost)
 	r.HandleFunc("/v1/users/login", Serviceuser.LoginUsers).Methods(http.MethodPost)
-	r.HandleFunc("/v1/logout", func(w http.ResponseWriter, r *http.Request) {
-		Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.LogoutUser)).ServeHTTP(w, r)
+	http.Handle("/v1/users/logout", Serviceuser.AuthHandler(http.HandlerFunc(Serviceuser.LogoutUser)))
+	r.HandleFunc("/v1/users/", func(w http.ResponseWriter, r *http.Request) {
```
Comment (bunin) [inline internal/server/server.go:21]: if I called `GET /v1/users/` I would expect to get a list of users

File: internal/users/getCurrentUser.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	err := pg.DB.QueryRow(query, userID).Scan(
+		&user.Name, &user.BirthDate, &user.Bio, &user.Location, &user.Nickname,
+		&subscriptionCount, &followerCount,
+	)
+	if err != nil {
+		services.ReturnErr(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	user.Followers = subscriptionCount
+	user.Following = followerCount
```
Comment (bunin) [inline internal/users/getCurrentUser.go:43]: why using additional variables only for these two? why not juts scan into `&user.Followers` and `&user.Following` as you did for the rest of the fields?

File: internal/users/usersStr.go [hunk 5]
```diff
@@ -20,6 +20,15 @@ type Users struct {
 	Following []int
 	Followers []int
 }
+type GetCurrentUser struct {
```
Comment (bunin) [inline internal/users/usersStr.go:23]: is it a request or a response?

File: internal/users/getCurrentUser.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	FROM
+		users_tweeter u
+	LEFT JOIN
+		followers_subscriptions fs1 ON u.id = fs1.follower_id
+	LEFT JOIN
+		followers_subscriptions fs2 ON u.id = fs2.subscription_id
+	WHERE
+		u.id = $1
+	GROUP BY
+		u.id, u.name, u.birthdate, u.bio, u.location, u.nickname
```
Comment (bunin) [inline internal/users/getCurrentUser.go:31]: this would need some improvements

**Review by bunin — APPROVED** [review]

File: internal/users/usersStr.go (hunk 5 above)
Comment (pm888) [reply internal/users/usersStr.go:23]: done

File: internal/users/getCurrentUser.go (hunk 4 above)
Comment (pm888) [reply internal/users/getCurrentUser.go:43]: done

File: internal/users/usersStr.go (hunk 5 above)
Comment (pm888) [reply internal/users/usersStr.go:23]: response

File: internal/users/getCurrentUser.go (hunk 4 above)
Comment (bunin) [reply internal/users/getCurrentUser.go:43]: done where? I don't see any changes here

File: internal/users/usersStr.go (hunk 5 above)
Comment (bunin) [reply internal/users/usersStr.go:23]: then please call it "*Response"

**Pull request merged**